flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractFetcherTest.java

```diff
@@ -108,13 +112,13 @@ public void testSkipCorruptedRecord() throws Exception {
 
 		final KafkaTopicPartitionState<Object> partitionStateHolder = fetcher.subscribedPartitionStates().get(0);
 
-		fetcher.emitRecord(1L, partitionStateHolder, 1L);
-		fetcher.emitRecord(2L, partitionStateHolder, 2L);
+		emitRecord(fetcher, 1L, partitionStateHolder, 1L);
```

Comment:
The changes in this file are purely an orthogonal refactoring, right? Could you put these in a separate commit, along with the removal of the `emitRecord()` method on the fetcher that is only used in tests?

Reply:
There are two related changes:
* I changed a bit the way no results are emitted (null before/empty queue now)
* I changed one test case slightly to emit multiple results in a single `emitRecordsWithTimestamps`

I think after extracting the refactoring to a separate commit it is visible more clearly.